src/proc/index.rs

```diff
@@ -140,7 +144,13 @@ impl BoundsCheckPolicies {
         types: &UniqueArena<crate::Type>,
         info: &valid::FunctionInfo,
     ) -> BoundsCheckPolicy {
-        match info[access].ty.inner_with(types).pointer_space() {
+        let ty = info[access].ty.inner_with(types);
+
+        if let crate::TypeInner::BindingArray { .. } = *ty {
+            return self.binding_array;
+        }
+
+        match ty.pointer_space() {
```

Comment:
This is fine (especially since the SPIR-V back end was already passing the base, not the access itself), but please rename `access` to `base`, and change the docs to say:

```
    /// Determine which policy applies to an indexed access of `base`.
    ///
    /// `base` is the base of an `Access` or `AccessIndex` expression,
    /// either a pointer to a value, or a value being indexed directly.
    ///
    /// See the documentation for [`BoundsCheckPolicy`] for details about
    /// when each policy applies.
```